### PR TITLE
6071 - ET - Fix server inactive session timeout

### PR DESCRIPTION
**Pull request by etelmon**
FIx inactive session timeout.
The session will time out on the server side even if the app tab or browser is closed (workstation turned off), network is disconnected for the duration of the inactive timeout.

Run 
`npm test`

### CODE REVIEW

File: src/services/auth.service.js [hunk 1]
```diff
@@ -425,6 +413,15 @@ function authProvider() {
         }
     };
 
+    function notifyUserActivityToBackend(socket) {
+        const lastNotif = Number(localStorage.lastNu || 0);
```
Comment (Jacob-Friesen) [inline src/services/auth.service.js:417]: This should be `Number(localStorage.lastNu) || 0;` instead if the intent is to handle `localStorage.lastNu` being a non number non falsely value. Also, something like `localStorage.lastNotifyUser` would be clearer for a name.

File: test/specs/auth.service.specs.js [hunk 2]
```diff
@@ -304,7 +304,7 @@ describe('Unit testing for auth,', () => {
             function(event, data, callback) {
                 console.log('emiting ' + event);
                 if (socketListeners[event]) {
-                    var r = socketListeners[event](data);
+                    var r = socketListeners[event](data, () => {});
```
Comment (Jacob-Friesen) [inline test/specs/auth.service.specs.js:307]: There should be new tests/more test modifications for this functionality since it is a major change.

**Review by Jacob-Friesen — CHANGES_REQUESTED** [review]
There is a code question and a testing one.

File: test/specs/auth.service.specs.js (hunk 2 above)
Comment (etelmon) [reply test/specs/auth.service.specs.js:307]: If you remove this, the test will not pass.

File: src/services/auth.service.js (hunk 1 above)
Comment (etelmon) [reply src/services/auth.service.js:417]: This is minimum code. Functionality would crash if the value is not correct. but the localStorage value is trusted as it is only set by the application itself, not a human.

On purpose the name of the var is unclear -> Hackers love meaningful variable in the localStorage!

Comment (Jacob-Friesen) [reply src/services/auth.service.js:417]: ok. As explained though the naming doesn't really help for security. Security needs to be backend based.

File: test/specs/auth.service.specs.js (hunk 2 above)
Comment (Jacob-Friesen) [reply test/specs/auth.service.specs.js:307]: PR is https://github.com/zimit-io/zimit-webapp/pull/4837

**Review by Jacob-Friesen — APPROVED** [review]

**Pull request merged**